src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -3418,6 +3451,10 @@ mongoc_collection_find_and_modify_with_opts (
       bson_append_document (&parts.extra, "let", 3, &appended_opts.let);
    }
 
+   if (&appended_opts.comment.value_type != BSON_TYPE_EOD) {
```

Comment:
```suggestion
   if (appended_opts.comment.value_type != BSON_TYPE_EOD) {
```

This may resolve errors reported on Windows builds.